biogeochem/EDMortalityFunctionsMod.F90

```diff
@@ -166,24 +169,44 @@ subroutine mortality_rates( cohort_in,bc_in,btran_ft, mean_temp,             &
        ! Carbon Starvation induced mortality.
        if ( cohort_in%dbh  >  0._r8 ) then
 
-          ! We compare storage with leaf biomass if plant were fully flushed, otherwise
-          ! mortality would be underestimated for plants that lost all leaves and have no
-          ! storage to flush new ones.
-          ! MLO. Why isn't this comparing with storage allometry (i.e., accounting for
-          !      cushion)?
+          ! Find the current ratio between storage biomass and leaf biomass, which will be
+          ! used to define carbon starvation mortality.  The reference leaf biomass is 
+          ! always for when plants are fully flushed (but accounting for damage and 
+          ! trimming).
           call bleaf(cohort_in%dbh,cohort_in%pft,cohort_in%crowndamage,cohort_in%canopy_trim, &
                1.0_r8, target_leaf_c)
           store_c = cohort_in%prt%GetState(store_organ,carbon12_element)
-
           call storage_fraction_of_target(target_leaf_c, store_c, frac)
-          if( frac .lt. 1._r8) then
-             cmort = max(0.0_r8,EDPftvarcon_inst%mort_scalar_cstarvation(cohort_in%pft) * &
-                  (1.0_r8 - frac))
-          else
+
+          ! Select the carbon starvation mortality model (linear or exponential)s.
+          select case (mort_cstarvation_model)
+          case (cstarvation_model_lin)
+             ! Linear model. Carbon starvation mortality will be zero when fraction of
+             ! storage is greater than or equal to mort_upthresh_cstarvation, and will
+             ! increase to the maximum mortality (mort_scalar_cstarvation) when frac = 0.
+             cmort = EDPftvarcon_inst%mort_scalar_cstarvation(cohort_in%pft) * &
+                max(0.0_r8, (EDPftvarcon_inst%mort_upthresh_cstarvation(cohort_in%pft)-frac) / &
+                            EDPftvarcon_inst%mort_upthresh_cstarvation(cohort_in%pft) )
+
+          case (cstarvation_model_exp)
```

Comment:
This also applies to your allometry PR @mpaiao , but we should start talking about expanding our regression tests to capture these different parameter mode settings.  As of the seed-dispersal PR by @glemieux , we can now do this.  Since this PR does have a change to the  parameter file, it will need to come in with a host-side update to the default parameter file. We can add the test in with that PR. Maybe we will integrate this PR and the allometry PR in sequence, and/or maybe make a combo tag of the two, and update the ctsm/e3sm side default files and their tests together.